src/event/event.controller.ts

```diff
@@ -76,14 +85,73 @@ export class EventController {
 
   @Get(":id/attendees")
   async getAttendees(
-    @Param('id') id: string
+    @Param('id') id: number
+  ): Promise<AttendeeWithUser[]> {
+    const allAttendeeObjects = await this.attendeeService.getAttendeesByEventId(id)
+    return allAttendeeObjects.map(fullAttendeeObject => ({
+      id: fullAttendeeObject.id,
+      user: {
+        id: fullAttendeeObject.user.id,
+        username: fullAttendeeObject.user.username
+      },
+    }))
+  }
+
+  @Post(":id/register")
+  async createAttendee(
+    @Param('id') id: number,
+    @Body() body: RegisterAttendeeBody
+  ): Promise<AttendeeWithUser> {
+    if (!body.hasOwnProperty("user")) {
```

Comment:
this validations (and all of the others) you can easily use JOI with nestjs and works really good